src/graph/graphcontext.c

```diff
@@ -107,15 +107,10 @@ void GraphContext_Release(GraphContext *gc) {
 void GraphContext_MarkWriter(RedisModuleCtx *ctx, GraphContext *gc) {
 	RedisModuleString *graphID = RedisModule_CreateString(ctx, gc->graph_name, strlen(gc->graph_name));
 
-	// Reopen only if key exists (do not re-create) make sure key still exists.
-	RedisModuleKey *key = RedisModule_OpenKey(ctx, graphID, REDISMODULE_READ);
-	if(RedisModule_KeyType(key) == REDISMODULE_KEYTYPE_EMPTY) goto cleanup;
-
-	// Mark as writer.
-	key = RedisModule_OpenKey(ctx, graphID, REDISMODULE_WRITE);
```

Comment:
Fixing the memory leak caused by not closing the key before here is actually the most important fix in this PR.

If you prefer the previous logic that first opens in read mode, we can just close the key before reopening, but I do not believe that doing so serves a purpose.